src/coreclr/jit/codegenxarch.cpp

```diff
@@ -5778,9 +5805,10 @@ void CodeGen::genCall(GenTreeCall* call)
     // To limit code size increase impact: we only issue VZEROUPPER before PInvoke call, not issue
     // VZEROUPPER after PInvoke call because transition penalty from legacy SSE to AVX only happens
     // when there's preceding 256-bit AVX to legacy SSE transition penalty.
-    if (call->IsPInvoke() && (call->gtCallType == CT_USER_FUNC) && GetEmitter()->Contains256bitAVX())
+    // This applies to 512bit AVX512 instructions as well.
+    if (call->IsPInvoke() && (call->gtCallType == CT_USER_FUNC) && (GetEmitter()->Contains256bitOrMoreAVX()))
     {
-        assert(compiler->canUseVexEncoding());
+        assert(GetEmitter()->Contains256bitOrMoreAVX() && compiler->canUseVexEncoding());
```

Comment:
First check is redundant with the check from the `if`
```suggestion
        assert(compiler->canUseVexEncoding());
```

Reply:
Done

Reply:
Looks like the latest sources don't have this one resolved yet.

Reply:
Nevermind. GitHub just hadn't updated for some reason.

Reply:
I just reverted the last commit - review changes to run CI again while I look at the superpmi  fail locally. 

If I remember right. superpmi were all passing last week- trying to figure out what changed outside of me rebasing to latest + possibly the last commit messing something up